pkg/operation/botanist/wanted_secrets.go

```diff
@@ -0,0 +1,718 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"fmt"
+	"net"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/features"
+	gardenletfeatures "github.com/gardener/gardener/pkg/gardenlet/features"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+var basicAuthSecretAPIServer = &secrets.BasicAuthSecretConfig{
```

Comment:
Hmm, I tried to enable basic auth for newly created Shoot but the kube-apiserver failed to start with:
```
  Warning  FailedMount  7s (x15 over 14m)  kubelet, ip-10-250-11-255.eu-west-1.compute.internal  MountVolume.SetUp failed for volume "kube-apiserver-basic-auth" : secret "kube-apiserver-basic-auth" not found
```

Is this reproducible also for you?

Reply:
Missed adding it in the deploy step. Thanks for finding it out. Couldn't complete all the tests as my dev setup has problems currently.